core/json/json_path.pest

```diff
@@ -1,7 +1,10 @@
 negative_index_indicator = ${ "#-" }
 array_offset = ${ ASCII_DIGIT+ }
 array_locator = ${ "[" ~ negative_index_indicator? ~ array_offset ~ "]" }
+relaxed_array_locator = ${ negative_index_indicator? ~ array_offset }
 
 root = ${ "$" }
 json_path_key = ${ identifier | string }
 path = ${ SOI ~ root ~ (array_locator | "." ~ json_path_key)* ~ EOI }
+
+relaxed_path = ${ SOI ~ ((root ~ (array_locator | "." ~ json_path_key)*) | json_path_key | relaxed_array_locator) ~ EOI }
```

Comment:
There is no need for fancy grammar, pass `OwnedValue` to `json_path` function:

1. If it's an integer, return `vec![Root, Index]`
2. If it's a string not starting with `$`, return `vec![Root, Key]`
3. If it starts with `$`, treat it as a regular path